climada/hazard/base.py

```diff
@@ -369,6 +371,104 @@ def set_vector(self, *args, **kwargs):
                        "Use Hazard.from_vector instead.")
         self.__dict__ = Hazard.from_vector(*args, **kwargs).__dict__
 
+    @classmethod
+    def from_raster_netcdf(
+        cls,
+        data: Union[xr.Dataset, str],
+        *,
+        intensity: str = "intensity",
+        fraction: Union[str, Callable[[np.ndarray], np.ndarray]] = "fraction",
+        coordinate_vars: Optional[Dict[str, str]] = None,
+    ):
+        """Read raster-like data from a NetCDF file
+
+        Parameters
+        ----------
+        data : xarray.Dataset or str
+            The NetCDF data to read from. May be a opened dataset or a path to a NetCDF
+            file, in which case the file is opened first.
+        intensity : str
+            Identifier of the `xarray.DataArray` containing the hazard intensity data.
+        fraction : str or Callable
+            Identifier of the `xarray.DataArray` containing the hazard fraction data.
+            May be a callable, in which case the callable is applied on the respective
+            intensity value to yield a fraction.
+        coordinate_vars : dict(str, str)
+            Mapping from default coordinate names to coordinate names used in the data
+            to read. The default coordinates are `time`, `longitude`, and `latitude`.
+
+        Returns
+        -------
+        hazard : climada.Hazard
+            A hazard object created from the input data
+
+        """
+        # If the data is a string, open the respective file
+        if not isinstance(data, xr.Dataset):
+            data = xr.open_dataset(data)
+        hazard = cls()  # TODO: Hazard type
+
+        # Update coordinate identifiers
+        coords = {"time": "time", "longitude": "longitude", "latitude": "latitude"}
+        if coordinate_vars is not None:
+            unknown_coords = [co not in coords for co in coordinate_vars]
+            if any(unknown_coords):
+                raise ValueError(
+                    f"Unknown coordinates passed: '{list(coordinate_vars.keys())}'. "
```

Comment:
Maybe list the unknown coordinates?

Reply:
Yeah, that sounds like a very reasonable idea 😅

Reply:
Done